src/app/modules/main/browser_section/provider/controller.nim

```diff
@@ -44,8 +45,8 @@ method getCurrentNetworkDetails*(self: Controller): NetworkDetails =
 method disconnect*(self: Controller) =
   discard self.dappPermissionsService.revoke("web3".toPermission())
 
-method postMessage*(self: Controller, message: string): string =
-  return self.providerService.postMessage(message)
+method postMessage*(self: Controller, requestType: string, message: JsonNode): string =
```

Comment:
do we have to use JsonNode here? this is similar as any and doesn't help much. (The whole callstack use JsonNode, and the question is I think valid till we do the status-go call)

Reply:
No preferences tbh. I'll change it to string

Reply:
wait, this is not a string right? What I mean there is probably some structure behind?

Reply:
yes, there's a json with a structure defined in the web3provider module of status-go, however the only item of that structure that nim should be aware of is the password

Reply:
nvm then if you use string or jsonnode